calicoctl/commands/label.go

```diff
@@ -0,0 +1,142 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/docopt/docopt-go"
+	"github.com/projectcalico/calicoctl/calicoctl/commands/constants"
+	"github.com/projectcalico/calicoctl/calicoctl/resourcemgr"
+)
+
+func Label(args []string) {
+	doc := constants.DatastoreIntro + `Usage:
+  calicoctl label <KIND> <NAME> <key>=<value> 
+  [--config=<CONFIG>] [--namespace=<NS>] [--overwrite]
+
+Examples:
+  # label a workload endpoint
+  calicoctl label workloadendpoints nginx --namespace=default app=web
+  # label a node and overwrite the original value of key 'cluster'
+  calicoctl label nodes node1 cluster=frontend --overwrite
+  # remove labels with key 'cluster' of the node
+  calicoctl label nodes node1 cluster= --overwrite
+
+Options:
+  -h --help                    Show this screen.
+  -c --config=<CONFIG>         Path to the file containing connection
+                               configuration in YAML or JSON format.
+                               [default: ` + constants.DefaultConfigPath + `]
+  -n --namespace=<NS>          Namespace of the resource.
+                               Only applicable to NetworkPolicy and WorkloadEndpoint.
+                               Uses the default namespace if not specified.
+  --overwrite                  If true, overwrite the value when the key is already
+                               present in labels. Otherwise reports error when the
+                               labeled resource already have the key in its labels.
+
+Description:
+  The label command is used to add or update a label on a resource. Resource types
+  that can be labeled are:
+
+    * bgpConfiguration
+    * bgpPeer
+    * felixConfiguration
+    * globalNetworkPolicy
+    * globalNetworkSet
+    * hostEndpoint
+    * ipPool
+    * networkPolicy
+    * node
+    * profile
+    * workloadEndpoint
+
+  The resource type is case insensitive and may be pluralized.
+
+  Attempting to label resources that are not exist will get an error.
+
+  When label a resource on an existing key in its labels:
+  - gets an error if option --overwrite is not provided.
+  - value of the key updates to specified value if option --overwrite is provided. 
+  `
+
+	parsedArgs, err := docopt.Parse(doc, args, true, "", false, false)
+	if err != nil {
+		fmt.Printf("Invalid option: 'calicoctl %s'. Use flag '--help' to read about a specific subcommand.\n", strings.Join(args, " "))
+		os.Exit(1)
+	}
+	if len(parsedArgs) == 0 {
+		return
+	}
+
+	fmt.Printf("parse args: %+v\n", parsedArgs)
```

Comment:
We use `github.com/sirupsen/logrus` for our logging. Please log out the args (on debug level) rather than printing to stdout

Reply:
sorry, did not delete it before commiting